### PR TITLE
Check for missing realizations

### PR DESCRIPTION
**Pull request by oyvindeide**
Checks which realizations are active before validating to account for failed realizations

### CODE REVIEW

File: python/res/enkf/export/summary_collector.py [hunk 1]
```diff
@@ -9,9 +9,9 @@
 class SummaryCollector(object):
 
     @staticmethod
-    def createActiveList(ert, fs):
+    def createActiveList(ensamble_size, fs):
```
Comment (markusdregi) [inline python/res/enkf/export/summary_collector.py]: Unless you have a very good reason for doing so, we should not change the current API...

File: python/res/enkf/jobs/scaling/scaling_job.py [hunk 2]
```diff
@@ -219,7 +220,8 @@ def has_data(observations, keys, ensamble_size, storage):
     Checks that all keys have data and returns a list of error messages
     """
     error_msg = "Key: {} has no data"
-    active_mask = BoolVector(True, ensamble_size)
+    active_realizations = SummaryCollector.createActiveList(ensamble_size, storage)
```
Comment (markusdregi) [inline python/res/enkf/jobs/scaling/scaling_job.py]: It seems a bit arbitrary to get this functionality from the `SummaryCollector`. How about using `storage.realizationList(RealizationStateEnum.STATE_HAS_DATA)` instead?

Comment (oyvindeide) [reply python/res/enkf/jobs/scaling/scaling_job.py]: That makes more sense :+1:

File: python/res/enkf/export/summary_collector.py (hunk 1 above)
Comment (oyvindeide) [reply python/res/enkf/export/summary_collector.py]: There is a reason, if it's good enough is another matter. It's symptomatic for a lot of these functions to take in EnkFMain and use a long chain of commands on it, making it very hard to create new functionality without pulling EnkFMain down to the lowest level, when in this case what was actually needed was an int. Changing that as the functionality was used seemed a rational way to start stabbing at the problem, but might not be worth a breaking change.

Comment (markusdregi) [reply python/res/enkf/export/summary_collector.py]: I see your reasoning, but let us save this for later... We should first make an evaluation of what we want the API to look like before we start making micro improvements and at the same time breaking our API.

File: python/res/enkf/jobs/scaling/scaling_job.py [hunk 3]
```diff
@@ -219,7 +219,8 @@ def has_data(observations, keys, ensamble_size, storage):
     Checks that all keys have data and returns a list of error messages
     """
     error_msg = "Key: {} has no data"
-    active_mask = BoolVector(True, ensamble_size)
+    active_realizations = storage.realizationList(RealizationStateEnum.STATE_HAS_DATA)
+    active_mask = BoolVector.createFromList(ensamble_size, active_realizations)
```
Comment (markusdregi) [inline python/res/enkf/jobs/scaling/scaling_job.py:227]: Could I have a test that fails on the previous setup and now succeeds? :)

File: python/res/enkf/export/summary_collector.py (hunk 1 above)
Comment (oyvindeide) [reply python/res/enkf/export/summary_collector.py]: :+1:

File: python/res/enkf/jobs/scaling/scaling_job.py (hunk 3 above)
Comment (oyvindeide) [reply python/res/enkf/jobs/scaling/scaling_job.py:227]: :+1:

File: python/tests/res/enkf/jobs/scaling/test_enkf_scaling_job.py [hunk 4]
```diff
@@ -708,3 +708,23 @@ def test_main_entry_point_block_data_calc():
...
+    shutil.copytree(test_data_dir, "test_data")
+    os.chdir(os.path.join("test_data"))
+
+    res_config = ResConfig("mini_fail_config")
+    ert = EnKFMain(res_config)
+
+    fs = ert.getEnkfFsManager().getCurrentFileSystem()
+    observations = ert.getObservations()
+
+    assert scaling_job.has_data(observations, ["GEN_PERLIN_1"], ert.getEnsembleSize(), ert.getEnkfFsManager().getCurrentFileSystem()) == []
```
Comment (markusdregi) [inline python/tests/res/enkf/jobs/scaling/test_enkf_scaling_job.py]: When GitHub activates the horizontal scrolling bar is a good hint that you should perhaps break your line ;)

Comment (oyvindeide) [reply python/tests/res/enkf/jobs/scaling/test_enkf_scaling_job.py]: Yes, still cleaning that up a little bit, also realized that this allows an ensemble with no data to pass, which it should not, so adding a test for that and a fix.

**Review by markusdregi — APPROVED** [review]
Looks good to me :+1: 

Made some issues with further work for this module...

**Pull request merged**